### PR TITLE
CA-381212, CA-381664, CA-381215, CP-44618

### PR DESCRIPTION
**Pull request by kc284**

### CODE REVIEW

**Comment by kc284** [comment]
I added one more commit. It's irrelevant to updates from CDN, but it was needed because the  space requiremetns for the HA hearbeat SR have now increased and the info was outdated. I removed it completely, so we don't have to update it again. The user can always find more details in the docs.

**Review by CitrixChris — APPROVED** [review]

File: XenModel/Actions/Updates/CheckForCdnUpdatesAction.cs [hunk 1]
```diff
@@ -67,15 +78,39 @@ protected override void Run()
...
+                             "no_repository_enabled or invalid_updateinfo_xml or invalid_repomd_xml. " +
+                             "See the server side logs for details.");
+                }
+                else if (ex.Message.Contains("Received error code HTTP/1.1 500 Internal Server Error\r\n from the server"))
+                {
+                    log.Warn("Failed to retrieve available updates.");
+                    log.Warn(ex.Message);
+                    log.Warn("One of the following errors has occurred: " +
+                             "get_host_updates_failed or updates_require_sync. " +
+                             "See the server side logs for details.");
```
Comment (danilo-delbusso) [inline XenModel/Actions/Updates/CheckForCdnUpdatesAction.cs:109]: should we avoid listing errors? they could change in the future. I'd just leave "see the server side logs for details". same for 404s above

**Review by danilo-delbusso — CHANGES_REQUESTED** [review]

File: XenModel/Actions/Updates/CheckForCdnUpdatesAction.cs (hunk 1 above)
Comment (kc284) [reply XenModel/Actions/Updates/CheckForCdnUpdatesAction.cs:109]: I have raised CP-44822 for a proper fix, so I epxect this to change aagian at some point. Until then, I only listed them lest someone complains the application logs show nothing. If you think it is not necessary, I can remove them, up to you.

Comment (danilo-delbusso) [reply XenModel/Actions/Updates/CheckForCdnUpdatesAction.cs:109]: >  I only listed them lest someone complains the application logs show nothing.

I think the warnings in general are very helpful and we should have them in the XC logs. I just want to remove the list of possible errors. They could become outdated and there's no way we'll remember to update them. Let's just point people to xensource.log

**Review by danilo-delbusso — APPROVED** [review]

**Pull request merged**